### DISCUSSION TITLE
How to configure jgit.dirtyWorkingTree/jgit.ignore without warnings with Maven 3.9.2

### DISCUSSION THREAD

**Discussion by merks** [eclipse-tycho/tycho#2404] [Q&A] [answered]
I get a great many of these warnings, but there seems to be no way to get rid of them:

> [INFO] ---------------------------[ eclipse-plugin ]---------------------------
> [WARNING] Parameter 'jgit.dirtyWorkingTree' is unknown for plugin 'tycho-packaging-plugin:3.0.4:build-qualifier (default-build-qualifier)'
> [WARNING] Parameter 'jgit.ignore' is unknown for plugin 'tycho-packaging-plugin:3.0.4:build-qualifier (default-build-qualifier)'
> [WARNING] Parameter 'jgit.dirtyWorkingTree' is unknown for plugin 'tycho-packaging-plugin:3.0.4:validate-id (default-validate-id)'
> [WARNING] Parameter 'jgit.ignore' is unknown for plugin 'tycho-packaging-plugin:3.0.4:validate-id (default-validate-id)'
> [WARNING] Parameter 'jgit.dirtyWorkingTree' is unknown for plugin 'tycho-packaging-plugin:3.0.4:validate-version (default-validate-version)'
> [WARNING] Parameter 'jgit.ignore' is unknown for plugin 'tycho-packaging-plugin:3.0.4:validate-version (default-validate-version)'
> [WARNING] Parameter 'jgit.dirtyWorkingTree' is unknown for plugin 'tycho-packaging-plugin:3.0.4:update-consumer-pom (default-update-consumer-pom)'
> [WARNING] Parameter 'jgit.ignore' is unknown for plugin 'tycho-packaging-plugin:3.0.4:update-consumer-pom (default-update-consumer-pom)'
> [WARNING] Parameter 'jgit.dirtyWorkingTree' is unknown for plugin 'tycho-packaging-plugin:3.0.4:package-plugin (default-package-plugin)'
> [WARNING] Parameter 'jgit.ignore' is unknown for plugin 'tycho-packaging-plugin:3.0.4:package-plugin (default-package-plugin)'

It comes from this configuration:

![image](https://github.com/eclipse-tycho/tycho/assets/208716/9b0c7284-2e59-4675-b0dd-bc6a3383dce9)

This worked without warnings with Maven 3.8.x.  I know the configuration actually takes effect because if I change warning to error the build fails because the tree is dirty.

How can we configure this without all these warnings?

**Comment by laeubi** [comment, marked as answer]
> How can we configure this without all these warnings?

This is currently not possible because it is new behavior of Maven 3.9.x, I have some ideas how to archive this in a different way but this needs a new concept of how to handle such configuration parameters:

- https://github.com/eclipse-tycho/tycho/issues/2080

**Reply by merks** [reply]
That’s what I suspected. Bummer. Thanks for the quick answer.

**Comment by guw** [comment]
I came across the same issue today. It looks like the warnings are generated for specific *executions*.
```
[WARNING] Parameter 'jgit.dirtyWorkingTree' is unknown for plugin 'tycho-packaging-plugin:3.0.4:build-qualifier (default-build-qualifier)'
[WARNING] Parameter 'jgit.dirtyWorkingTree' is unknown for plugin 'tycho-packaging-plugin:3.0.4:validate-id (default-validate-id)'
[WARNING] Parameter 'jgit.dirtyWorkingTree' is unknown for plugin 'tycho-packaging-plugin:3.0.4:validate-version (default-validate-version)'
[WARNING] Parameter 'jgit.dirtyWorkingTree' is unknown for plugin 'tycho-packaging-plugin:3.0.4:update-consumer-pom (default-update-consumer-pom)'
[WARNING] Parameter 'jgit.dirtyWorkingTree' is unknown for plugin 'tycho-packaging-plugin:3.0.4:package-plugin (default-package-plugin)'
```

I haven't tested this yet myself but it is possible to  move `<configuration>` within an `<execution>`. Thus, it maybe necessary to move the `<jgit.dirtyWorkingTree>` into a specific `<execution>`. 

But it could also be the case that the `default-build-qualifier` execution should allow this option somehow.

**Reply by laeubi** [reply]
The JGit is used as an extension mechanism, but extensions can't contribute to the mojo configuration therefore maven complains because the mojo actually do not has such a parameter name.
